src/ucp/core/ucp_ep.c

```diff
@@ -232,19 +234,77 @@ ucs_status_t ucp_ep_create_base(ucp_worker_h worker, const char *peer_name,
         goto err_free_ep_control_ext;
     }
 
+    return ep;
+
+err_free_ep_control_ext:
+    ucs_free(ucp_ep_ext_control(ep));
+err_free_ep:
+    ucs_strided_alloc_put(&worker->ep_alloc, ep);
+err:
+    return NULL;
+}
+
+static int ucp_ep_shall_use_indirect_id(ucp_context_h context,
+                                        unsigned ep_init_flags)
+{
+    return !(ep_init_flags & UCP_EP_INIT_FLAG_INTERNAL) &&
+           ((context->config.ext.proto_indirect_id == UCS_CONFIG_ON) ||
+            ((context->config.ext.proto_indirect_id == UCS_CONFIG_AUTO) &&
+             (ep_init_flags & UCP_EP_INIT_ERR_MODE_PEER_FAILURE)));
+}
+
+ucs_status_t ucp_ep_create_base(ucp_worker_h worker, unsigned ep_init_flags,
```

Comment:
why need to change this function?

Reply:
after moving `releasing ID`, `removing EP keepalive`, `deleting EP from worker's ep_list` to `ucp_ep_destroy_base`.
I wanted to align `ucp_ep_create_base` and `ucp_ep_destroy_base`.

Reply:
why aligning ep create and destroy is essential for fixing the removal of ep from worker list?
it could be done in next PR?

Reply:
yes, we could do this in the next PR to simplify the current one.

Reply:
removed changes for `ucp_ep_create_base`